openlcsd/flow/tasks.py

```diff
@@ -641,9 +641,7 @@ def deduplicate_source(context, engine):
                             os.remove(path)
                 else:
                     swhids = list(set(swhids))
-
-                if swhids:
-                    context['source_info']['swhids'] = swhids
+                context['source_info']['swhids'] = swhids if swhids else []
```

Comment:
Does that mean it allowed the swhids is None? When it will be None?

Reply:
the existing code means if swhids exist, will set `context['source_info']['swhids'] = swhids`, but if not exist, will not be set, but in the backend, if not set, we will get `None` for swhids.

code:
https://github.com/RedHatProductSecurity/openlcs/blob/69b18fa2c290bc9358de7d3d7e58b2b69818d35f/openlcs/packages/views.py#L807